src/main/java/teammates/common/util/ActivityLogGenerator.java

```diff
@@ -0,0 +1,329 @@
+    }
+    
+    private String generateLogIdWithoutGoogleId(Map<String, String[]> params, long time) {
+        String courseId = HttpRequestHelper.getValueFromParamMap(params, Const.ParamsNames.COURSE_ID);
+        String studentEmail = HttpRequestHelper.getValueFromParamMap(params, Const.ParamsNames.STUDENT_EMAIL);
+        if (courseId != null && studentEmail != null) {
+            return studentEmail + Const.ActivityLog.FIELD_CONNECTOR
+                   + courseId + Const.ActivityLog.FIELD_CONNECTOR
+                   + formatTimeForId(new Date(time));
+        }
+        return Const.ActivityLog.AUTH_NOT_LOGIN + Const.ActivityLog.FIELD_CONNECTOR + formatTimeForId(new Date(time));
+    }
+    
+    private String generateLogIdWithGoogleId(String googleId, long time) {
+        return googleId + Const.ActivityLog.FIELD_CONNECTOR + formatTimeForId(new Date(time));
+    }
+    
+    private String formatTimeForId(Date date) {
+        SimpleDateFormat sdf = new SimpleDateFormat(Const.ActivityLog.TIME_FORMAT_LOGID);
+        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+        return sdf.format(date.getTime());
+    }
+    
+    /**
+     * Generates {@link ActivityLogEntry} from {@link AppLogLine} provided by GAE.
+     * <p>
+     * If the log message in {@link AppLogLine} is not in the desired format specified in {@link ActivityLogEntry},
+     * an {@link ActivityLogEntry} will be returned with log message : 'Error. Problem parsing log message from the server.'
+     * 
+     * @param appLog
+     * @return {@link ActivityLogEntry}
+     */
+    public ActivityLogEntry generateActivityLogFromAppLogLine(AppLogLine appLog) {
+        try {
+            String[] tokens = appLog.getLogMessage().split(Pattern.quote(Const.ActivityLog.FIELD_SEPARATOR), -1);
+            return initActivityLogUsingAppLogMessage(appLog, tokens);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            return initActivityLogAsFailure(appLog, e);
+        }
+    }
+
+    private ActivityLogEntry initActivityLogAsFailure(AppLogLine appLog, ArrayIndexOutOfBoundsException e) {
+        Builder builder = new Builder(Const.ActivityLog.UNKNOWN, Const.ActivityLog.UNKNOWN, appLog.getTimeUsec());
+        // TODO : can use template here
+        String logMessage = "<span class=\"text-danger\">" + Const.ActivityLog.MESSAGE_ERROR_LOG_MESSAGE_FORMAT
+                            + "</span><br>System Error: " + e.getMessage() + "<br>" + appLog.getLogMessage();
+        builder.withLogMessage(logMessage);
+        return builder.build();
+    }
+
+    private ActivityLogEntry initActivityLogUsingAppLogMessage(AppLogLine appLog, String[] tokens) {
+        // TEAMMATESLOG|||ACTION_NAME|||ACTION_RESPONSE|||TO_SHOW|||ROLE|||NAME|||GOOGLE_ID|||EMAIL
+        // |||MESSAGE(IN HTML)|||URL|||TIME_TAKEN
+        String actionName = tokens[ActivityLogEntry.POSITION_OF_ACTION_NAME];
```

Comment:
I thought we could put everything to one place and make it consistent.

I think we should not expose `POSITION_OF_*` variable, I will change later.